### ISSUE TITLE
Tests started to fail (with newest Crypt::OpenSSL::RSA?)

### ISSUE THREAD

**Issue by eserte** [catalyst/Authen-NZRealMe#5]
On my smoker systems the test suite fails:
```
#   Failed test 'verified sigs without throwing exception'
#   at t/30-xml-sigs.t line 191.
#          got: 'SignedInfo block signature does not match at /home/cpansand/.cpan/build/2018052512/Authen-NZRealMe-1.16-W8UU4d/blib/lib/Authen/NZRealMe/XMLSig.pm line 435.
# '
#     expected: ''

#   Failed test 'verify method returned true'
#   at t/30-xml-sigs.t line 192.

#   Failed test 'with appropriate message'
#   at t/30-xml-sigs.t line 228.
#                   'SignedInfo block signature does not match at /home/cpansand/.cpan/build/2018052512/Authen-NZRealMe-1.16-W8UU4d/blib/lib/Authen/NZRealMe/XMLSig.pm line 435.
# '
#     doesn't match '(?^:Digest of signed element 'fourfivesix' differs from that given in reference block)'
# Looks like you failed 3 tests of 33.
t/30-xml-sigs.t ........... 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/33 subtests 
... (etc, more fails following) ...
```

Statistical analysis suggests that this happens with Crypt::OpenSSL::RSA 0.30, see http://analysis.cpantesters.org/solved?distv=Authen-NZRealMe-1.16#mod%3ACrypt%3A%3AOpenSSL%3A%3ARSA (@toddr FYI)

**Comment by toddr** [comment]
I highly suspect the root cause is https://github.com/toddr/Crypt-OpenSSL-RSA/commit/1149369250b667ff22df50c72a3fce4476636315. My guess would be that the code is hard coding its assumptions about what the hash will be. I'll look later tonight

**Comment by toddr** [comment]
This was fixed in toddr/Crypt-OpenSSL-RSA@65ff114cae2ed17abea12304d195614f5dec01dd and then released to metacpan as 0.31. You might want to depend on 0.31 now or otherwise you can close this as a non-issue for this module.

**Comment by grantm** [comment]
I did eventually get round to putting out a new release depending on 0.31 (along with other changes that needed to go out). It seems to be doing OK with CPAN testers. Thank you both for the initial report and followup analysis.
